### DISCUSSION TITLE
Support on git integration

### DISCUSSION THREAD

**Discussion by henrythebuilder** [lapce/lapce#1748] [Q&A]
Hi,
I found some problems in showing the git status via interface:
1) the status of git (git tab) is reported only if you open Lapce inside the root folder of git (where the '.git' folder resides), this happens both for the list of modified files and for the single modifications inside the files. When a workspace resides in a repository with more contents in multiple directories opening Lapce inside the workspace sub-folder does not display any repository changes.
2) in git mode (git tab) by selecting a modified file a 'diff' buffer is opened (correctly) but it seems complicated to go back to the original file unless by closing the 'diff' buffer and reopen the original modified file.

Is there a way to use 'Lapce' to workaround these problems or is it '_application_' problems?

Thanks in advance

Enrico

**Comment by claytonrcarter** [comment]
I noticed the same thing. I have a patch working locally to address the first item, where lapce is opened inside a subdirectory of a git repo. I'm still too new around here to know what to do about the second one though. 😄

**Comment by henrythebuilder** [comment]
Hi @claytonrcarter, I also tried to find the cause in the code.
The only thing that made me improve it was to use a different api to open the repository, `Repository::discover` (or maybe `Repository::open_ext`) instead of `Repository::open`  in order to detect a repository at the above path.
Replacing the original calls with the new api the list of modified files is correct, but other side effects have taken over for example in the '_diff view_' ... I continue the tests ...

Thanks to everyone who contributes to the development of Lapce' !!!

**Reply by claytonrcarter** [reply]
Yes I found a couple of other places where the project is assuming that the workspace correlates with the root of a git repo. I've patched most of them, but there are a couple remaining that I have to think about. For what it's worth I just checked and the diff view is working correctly. Or at least I think it is!
